doc.go

```diff
@@ -81,6 +81,16 @@
 //	 }
 //	 defer session.Close()
 //
+// By default, PasswordAuthenticator will attempt to authenticate regardless of what implementation the server returns
+// in its AUTHENTICATE message as its authenticator, (e.g. org.apache.cassandra.auth.PasswordAuthenticator).  If you
+// wish to restrict this you may use PasswordAuthenticator.AllowedAuthenticators:
```

Comment:
nit: Maybe a comment on the interface function itself could also be useful

Reply:
I was thinking about a comment on the type itself ( `conn.go` ) so that this information shows up on the docs section for the type (in pkg.go.dev) and when using an IDE:

```
type PasswordAuthenticator struct {
	Username                      string
	Password                       string

        // Setting this to nil or empty will allow any authenticator provided by the server
	AllowedAuthenticators []string
}
```

Reply:
:+1: sounds great, agree that this is the most appropriate place.  Will make that change.

Reply:
made an attempt to document both.  Thought it would be good to make it clear that the default behavior of other drivers is to allow any authenticator,  generally I don't think people should configure this, and the presence of documentation may lead them to think they need to do so, so felt it was good to clarify that.   Hopefully that looks ok.

Reply:
lgtm 👍

Reply:
Thanks, i'll squash the commits and update the patch by/reviewed by :+1:

Reply:
done, hopefully this is good to go!